sycl/source/detail/config.hpp

```diff
@@ -163,6 +165,118 @@ template <> class SYCLConfig<SYCL_PI_TRACE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_DEVICE_TRIPLE> {
+  using BaseT = SYCLConfigBase<SYCL_DEVICE_TRIPLE>;
+
+public:
+  static std::vector<triple> *get() {
+    static bool Initialized = false;
+    static std::string String;
+    static std::vector<triple> TripleList;
+
+    // Configuration parameters are processed only once, like reading a string
+    // from environment and converting it into a typed object.
+    if (Initialized) {
+      if (TripleList.size()) {
+        return &TripleList;
+      } else {
+        return nullptr;
+      }
+    }
+
+    const std::array<std::pair<std::string, info::device_type>, 5>
+        SyclDeviceTypeMap = {{{"host", info::device_type::host},
+                              {"cpu", info::device_type::cpu},
+                              {"gpu", info::device_type::gpu},
+                              {"acc", info::device_type::accelerator},
+                              {"*", info::device_type::all}}};
+    const std::array<std::pair<std::string, backend>, 4> SyclBeMap = {
+        {{"opencl", backend::opencl},
+         {"level0", backend::level_zero},
+         {"level_zero", backend::level_zero},
+         {"cuda", backend::cuda}}};
+
+    Initialized = true;
+    const char *ValStr = BaseT::getRawValue();
+    if (ValStr) {
+      String = ValStr;
+      std::transform(String.begin(), String.end(), String.begin(), ::tolower);
+      bool MoreTriple = true;
+      size_t Pos = 0;
+      while (MoreTriple) {
+        MoreTriple = false;
+        triple Trp;
+        // device_type is required entry
+        auto It = std::find_if(
+            std::begin(SyclDeviceTypeMap), std::end(SyclDeviceTypeMap),
+            [=,
+             &Pos](const std::pair<std::string, info::device_type> &element) {
+              size_t Found = String.find(element.first, Pos);
+              if (Found != std::string::npos) {
+                Pos = Found;
+                return true;
+              }
+              return false;
+            });
+        if (It == SyclDeviceTypeMap.end())
+          pi::die("Invalid device_type. "
+                  "Valid values are host/cpu/gpu/acc/*");
+
+        // initialize optional entries with default values
+        if (It->second == info::device_type::gpu) {
+          Trp = {It->second, backend::level_zero, DEVICE_NUM_UNSPECIFIED};
```

Comment:
Do we really need to do it? I would expect that specifying "gpu;*" I get OpenCL GPU, CUDA GPU, L0 GPU devices(if I have corresponding HW), but it seems this code breaks this logic.